From: =?utf-8?q?Jeremy_B=C3=ADcha?= <jbicha@ubuntu.com>
Date: Tue, 7 Oct 2025 19:06:53 -0400
Subject: tests: Don't run openjpeg tests on big-endian

This could mean that openjpeg is broken on these architectures!
---
 tests/check/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/check/meson.build b/tests/check/meson.build
index c5ea414..38e7b70 100644
--- a/tests/check/meson.build
+++ b/tests/check/meson.build
@@ -67,7 +67,7 @@ base_tests = [
   [['elements/nvenc.c'], false, [gstgl_dep, gmodule_dep]],
   [['elements/nvdec.c'], not gstgl_dep.found(), [gstgl_dep, gmodule_dep]],
   [['elements/svthevcenc.c'], not svthevcenc_dep.found(), [svthevcenc_dep]],
-   [['elements/openjpeg.c'], not openjpeg_dep.found(), [openjpeg_dep]],
+   [['elements/openjpeg.c'], not openjpeg_dep.found() or host_machine.endian() == 'big', [openjpeg_dep]],
   [['elements/pcapparse.c'], false, [libparser_dep]],
   [['elements/pnm.c'], get_option('pnm').disabled()],
   [['elements/proxysink.c'], get_option('proxy').disabled()],
